File: BOJ(Java 공부용)/1620.java
import java.io.*;
import java.util.HashMap;
import java.util.StringTokenizer;

public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static int[] gox = {0, 1, 0, -1};
    static int[] goy = {1, 0, -1, 0};

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken()), m = Integer.parseInt(st.nextToken());
        HashMap<Integer, String> map1 = new HashMap<>();
        HashMap<String, Integer> map2 = new HashMap<>();
        for (int i = 1; i <= n; i++) {
            String s = br.readLine();
            map1.put(i, s);
            map2.put(s, i);
        }
        for (int i = 0; i < m; i++) {
            String s = br.readLine();
            if (s.charAt(0) >= '1' && s.charAt(0) <= '9') {
                int a = Integer.parseInt(s);
                bw.write(map1.get(a) + "\n");
            } else {
                bw.write(map2.get(s) + "\n");
            }
        }
        bw.flush();
        bw.close();
    }
}

File: BOJ(Kotlin)/1620.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

fun main(args: Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val st = StringTokenizer(br.readLine())
    val n = st.nextToken().toInt()
    val m = st.nextToken().toInt()
    val map1 = mutableMapOf<Int,String>()
    val map2 = mutableMapOf<String,Int>()
    for (i in 1..n) {
        val s = br.readLine()
        map1[i] = s
        map2[s] = i
    }

    for (i in 0 until m) {
        val s = br.readLine()
        if(s[0] in '1'..'9') {
            bw.write(map1[s.toInt()]+"\n")
        }
        else {
            bw.write(map2[s].toString()+"\n")
        }
    }
    bw.flush()
    bw.close()
}